src/screens/LoginScreen.jsx

```diff
@@ -0,0 +1,64 @@
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from '../context/auth-context'
+import { useNavigate } from "react-router";
+import "./Authentication.css";
+
+const LoginScreen = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
+  const { login } = useAuth();
+  const navigate = useNavigate();
+  
+  
+  const clickHandler = async (e) => {
+    e.preventDefault();  
+    try {
+        await login(email, password);
+        navigate('/');
+    } catch (err) {
+        setError(err.message);
+    }
```

Comment:
You can do this in context, try to move `logic` in context if you are using just call the login here

Reply:
I have put the login/signup logic in context - Here i am calling that function and setting the error (if any) in this specific file.